transactions/src/main/java/org/ehcache/transactions/xa/internal/XAStore.java

```diff
@@ -115,7 +115,7 @@ public XAStore(Class<K> keyType, Class<V> valueType, Store<K, SoftLock<V>> under
     this.recoveryXaResource = new EhcacheXAResource<>(underlyingStore, journal, transactionContextFactory);
     this.eventSourceWrapper = new StoreEventSourceWrapper<>(underlyingStore.getStoreEventSource());
 
-    ContextManager.associate(underlyingStore).withParent(this);
+    this.statisticsService.registerWithParent(underlyingStore, this);
```

Comment:
All the other stores have their registration done in their provider... why is this one different?